docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+default       nginx-6757dd8769-k6h6x                   1/1       Running   0          2m
+default       nginx-6757dd8769-vzd5b                   1/1       Running   0          29m
+
+HPAS:
+
+NAMESPACE   NAME       REFERENCE          TARGETS       MINPODS   MAXPODS   REPLICAS   AGE
+default     nginxext   Deployment/nginx   30/9 (avg)   1         3         3          29m
+
+```
+
+Voila
+
+### Troubleshooting
+
+- Make sure you have the Aggregation layer and the certificates set up as per the requirements section.
+- Always make sure the metrics you want to autoscale on are available
+As you create the HPA, the DCA will parse the manifest and query Datadog to try to fetch the metric.
+If there is a typo or if the metric does not exist you will see an error such as:
+```
+2018-07-03 13:47:56 UTC | ERROR | (datadogexternal.go:45 in queryDatadogExternal) | Returned series slice empty
+```
+You can check in the Configmap used to store and share the HPA state by the DCA:
+`kubectl get cm datadog-hpa -o yaml`
+will yield:
+```
+apiVersion: v1
+data:
+  external.metrics.default.nginxext-nginx.net.request_per_s: '{"name":"nginx.net.request_per_s","labels":{"kube_container_name":"nginx"},"ts":1530625676,"hpa_name":"nginxext","hpa_namespace":"default","value":0,"valid":false}'
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2018-07-03T13:40:50Z
+  name: datadog-hpa
+  namespace: default
+  resourceVersion: "1742"
+```
+Here you can see if the metric has a typo or check that it exists in Datadog. If the metric's flag `Valid` is set to false, the metric will not be considered in the HPA pipeline.
+In this case, the scope could be incorrect, if we fix it by setting the right labels in the hpa manifest we will see:
+```
+data:
+  external.metrics.default.nginxext-nginx.net.request_per_s: '{"name":"nginx.net.request_per_s","labels":{"app":"puppet","env":"demo"},"ts":1530625976,"hpa_name":"nginxext","hpa_namespace":"default","value":2493,"valid":true}'
+```
+In the ConfigMap, indicating that it works.
+
+- If you see:
+```
+Conditions:
+  Type           Status  Reason                   Message
+  ----           ------  ------                   -------
+  AbleToScale    True    SucceededGetScale        the HPA controller was able to get the target's current scale
+  ScalingActive  False   FailedGetExternalMetric  the HPA was unable to compute the replica count: unable to get external metric default/nginx.net.request_per_s/&LabelSelector{MatchLabels:map[string]string{kube_container_name: nginx,},MatchExpressions:[],}: unable to fetch metrics from external metrics API: the server could not find the requested resource (get nginx.net.request_per_s.external.metrics.k8s.io)
+
+```
+
+Then it's likely that you don't have the proper RBAC set for the HPA.
+Make sure that `kubectl api-versions` shows:
+```
+autoscaling/v2beta1
+[...]
+external.metrics.k8s.io/v1beta1 
+```
+The latter will show up if the DCA properly registers as an External Metrics Provider.
```

Comment:
r/`The latter will show up if the DCA properly registers as an External Metrics Provider.`/`The latter shows up only if the DCA properly registers as an External Metrics Provider.`